services/brig/test/integration/API/Settings.hs

```diff
@@ -35,106 +35,138 @@ tests defOpts manager brig galley = return $ do
         "setEmailVisibility"
         [ testGroup
             "/users/"
-            [ testCase "EmailVisibleIfOnTeam"
-                . runHttpT manager
-                $ testUsersEmailVisibleIffExpected defOpts brig galley Opt.EmailVisibleIfOnTeam,
-              testCase "EmailVisibleToSelf"
-                . runHttpT manager
-                $ testUsersEmailVisibleIffExpected defOpts brig galley Opt.EmailVisibleToSelf
-            ],
+            $ ((,) <$> [minBound ..] <*> [minBound ..])
+              <&> \(viewingUserIs, visibility) -> do
+                testCase (show (viewingUserIs, visibility))
+                  . runHttpT manager
+                  $ testUsersEmailVisibleIffExpected defOpts brig galley viewingUserIs visibility,
           testGroup
             "/users/:uid"
-            [ testCase "EmailVisibleIfOnTeam"
-                . runHttpT manager
-                $ testGetUserEmailShowsEmailsIffExpected defOpts brig galley Opt.EmailVisibleIfOnTeam,
-              testCase "EmailVisibleToSelf"
-                . runHttpT manager
-                $ testGetUserEmailShowsEmailsIffExpected defOpts brig galley Opt.EmailVisibleToSelf
-            ]
+            $ ((,) <$> [minBound ..] <*> [minBound ..])
+              <&> \(viewingUserIs, visibility) -> do
+                testCase (show (viewingUserIs, visibility))
+                  . runHttpT manager
+                  $ testGetUserEmailShowsEmailsIffExpected defOpts brig galley viewingUserIs visibility
         ]
     ]
 
-data UserRelationship = SameTeam | DifferentTeam | NoTeam
+-- | The user looking at users is always a team creator; the user looked falls into the
+-- different categories enumerated here.
+data ViewedUserIs = SameTeam | DifferentTeam | NoTeam
 
-expectEmailVisible :: Opt.EmailVisibility -> UserRelationship -> Bool
-expectEmailVisible Opt.EmailVisibleIfOnTeam SameTeam = True
-expectEmailVisible Opt.EmailVisibleIfOnTeam DifferentTeam = True
-expectEmailVisible Opt.EmailVisibleIfOnTeam NoTeam = False
-expectEmailVisible Opt.EmailVisibleToSelf SameTeam = False
-expectEmailVisible Opt.EmailVisibleToSelf DifferentTeam = False
-expectEmailVisible Opt.EmailVisibleToSelf NoTeam = False
+-- | Analog of 'ViewedUserIs' for the viewing user.
+data ViewingUserIs = Creator | Member | Guest
+  deriving (Eq, Show, Enum, Bounded)
+
+expectEmailVisible :: Opt.EmailVisibility -> ViewingUserIs -> ViewedUserIs -> Bool
+expectEmailVisible Opt.EmailVisibleIfOnTeam = \case
+  Creator -> \case
+    SameTeam -> True
+    DifferentTeam -> True
+    NoTeam -> False
+  Member -> \case
```

Comment:
These cases could be de-duplicated like for `EmailVisibleToSelf` below, I think.